### ISSUE TITLE
Forces:  Ion-Ion forces from CoulombPBCAA are not correct.  

### ISSUE THREAD

**Issue by rcclay** [QMCPACK/qmcpack#2031]
During debugging of the ion-ion force bug in the CCZ estimator addressed in PR #2019, @Paul-St-Young found large disagreements between the fixed CCZ estimator+Quantum Espresso forces and the forces calculated by CoulombPBCAA.  CoulombPBCAA forces exhibit extremely slow convergence to a seemingly correct answer, which to me indicates a problem with the computation of the short-range piece of the force.  I will update this with some actual numerics and bare bones unit test, the lack of which had sort of slipped under the radar.

**Comment by tiihonej** [comment]
This has probably been done with divergent code from that in PR #2019, so I would review this issue after the merge is done.

What I get for IonIon forces with PR #2019 code, using QE for H2 with 2.0 bohr separation is the following, which looks fine to me, although the test is not very comprehensive:

```
			cutoff=50	cutoff=25	cutoff=15
QE			-0.240613385	-0.240613385	-0.240613385
ForceChiesaPBCAA	-0.2406133827	-0.2406133410	-0.2406395242
CoulombPBCAA		-0.24061338269	-0.24061334100	-0.24063952424
```

**Comment by rcclay** [comment]
Try this one, courtesy of Paul.  64 disordered H atoms in a box.  Discrepancies (if they exist) will be obvious with the 3 force components of the first atom.  I hope that my code path just diverged, but on the other hand, dimers are pretty special so it's not inconceivable that they might pass simple tests whereas a more complicated system wouldn't.  CCZ post bug fix is bang on.
[rs125-i5.zip](https://github.com/QMCPACK/qmcpack/files/3757081/rs125-i5.zip)

**Comment by rcclay** [comment]
I'll run the same example with your branch from PR #2019 as well.

**Comment by tiihonej** [comment]
Okay, I agree there's indeed a problem with CoulombPBCAA Ion-Ion forces, which only manifests with more than two ions. Hopefully it's easy to find and fix. Still, since CCZ seems to be fine, I'd merge PR #2019 now and work out a solution for this one separately. What do you think?

**Comment by rcclay** [comment]
Definitely merge PR#2019.  The issue was pre-existing.

**Comment by rcclay** [comment]
OK.  There's something goofy going on with the code.  Basically, the X components of all the forces come out right, but Y and Z are messed up in this test case.  However, there's no explicit reference made to the cartesian index, so who knows.  For now, I suggest using the simple loop structure of ForceChiesaPBCAA's ion force computation.  This gives the right answers, is simpler to read, and will fix this part of the code while I figure out the academically interesting question of wtf is going on with the current code.  If everyone is agreeable, I'll push up a fix and a unit test after PR #2019 is merged.

**Comment by tiihonej** [comment]
I barely missed your post, but I think I have already worked on and submitted more or less what you suggested. Actually I found similar strange behavior. Anyway, the problem was in CoulombPBCAA::evalSRwithForces. There might be something that doesn't work, when splint method is used on EwaldHandler3D, so I'm using sdDf now for forces (same as CCZ), and it seems to work.

**Comment by rcclay** [comment]
I don't think it's the spline, because porting over the code from Chiesa but using the dv/dr=rinv*(d_rv_dr-V) term (where d_rv_dr and V are evaluated from the splines) instead of srDf works fine.  This _feels_ like a slightly screwy distance table access.

**Comment by rcclay** [comment]
Closed by PR #2019
